volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -26,43 +26,113 @@
 
 
 class module(generic.GenericIntelProcess):
+    @property
+    def mod_mem_type(self):
+        """Return the mod_mem_type enum choices if available or None if not"""
+        if not self.has_member("_mod_mem_type"):
```

Comment:
This shouldn't be checking for existence, the `_mod_mem_type` should be defined as None in the initializer technically.  This should be a test of the value to determine if it's been set before or not.

Given that this is just looking up an enumeration, it *shouldn't* be necessary to ever look it up again so it might be inefficient to test for `is None` because we might rerun the check code even after we know it's not possible, however, since it's a dictionary, we can return an empty dictionary which won't pass an `is None` test (allowing the memoization to shortcircuit) but will fail an `if blah.mod_mem_type:` test.  Also, the memoized variable (`_mod_mem_type`) should be initialized in the initializer...